### PR TITLE
Message passing between drupal and camel.

### PR DESCRIPTION
**Pull request by ruebot**
@whikloj @DiegoPino 

Hopefully this looks just like https://github.com/Islandora-CLAW/CLAW/pull/138

### CODE REVIEW

File: camel/indexing/islandora-indexing-triplestore/pom.xml [hunk 1]
```diff
@@ -0,0 +1,112 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>ca.islandora.camel</groupId>
+    <artifactId>islandora-parent</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
+    <relativePath>../../pom.xml</relativePath>
```
Comment (whikloj) [inline camel/indexing/islandora-indexing-triplestore/pom.xml]: I don't think this is necessary, but we need to add this as a sub-module [here](https://github.com/Islandora-CLAW/CLAW/blob/sprint-002/camel/pom.xml#L38-L42)

Comment (whikloj) [inline camel/indexing/islandora-indexing-triplestore/pom.xml]: For historical reference. I was wrong. We do need this. 😭

File: install/scripts/islandora-karaf-components.sh [hunk 2]
```diff
@@ -9,3 +9,12 @@ fi
 
 echo "Installing Islandora Sync Gateway"
 $KARAF_CLIENT -f $KARAF_CONFIGS/islandora_sync_gateway.script
+sleep 10
+echo "Installing Islandora Basic Image Service"
+$KARAF_CLIENT -f $KARAF_CONFIGS/islandora_basic_image.script
+sleep 10
+echo "Installing Islandora Collection Service"
+$KARAF_CLIENT -f $KARAF_CONFIGS/islandora_collection.script
+sleep 10
```
Comment (whikloj) [inline install/scripts/islandora-karaf-components.sh]: I just realized that we don't need or want to load the old camel Basic Image service and Collection Service. We should only load the Sync Gateway and Triplestore Indexer.

File: install/configs/karaf/watch.script [hunk 3]
```diff
@@ -1,3 +1,6 @@
 bundle:watch --start mvn:ca.islandora.camel.component/islandora-camel-component/0.0.1-SNAPSHOT
 bundle:watch --start mvn:ca.islandora.camel.sync/islandora-sync-gateway/0.0.1-SNAPSHOT
+bundle:watch --start mvn:ca.islandora.camel.services/islandora-collection-service/0.0.1-SNAPSHOT
+bundle:watch --start mvn:ca.islandora.camel.services/islandora-basic-image-service/0.0.1-SNAPSHOT
```
Comment (whikloj) [inline install/configs/karaf/watch.script]: Also don't want to watch the collection-service or basic-image-service for changes.

**Comment by whikloj** [comment]
Also this breaks because I switched to php 5.6.20 and the php5-stomp package requires virtual package phpapi-20121212 which depends on php5-common 5.5.9. Ubuntu FTW.

**Comment by br2490** [comment]
What will happen on 16.04 with PHP7?

On Wed, Apr 27, 2016 at 12:20 PM, Jared Whiklo <notifications@github.com>
wrote:

> Also this breaks because I switched to php 5.6.20 and the php5-stomp
> package requires virtual package phpapi-20121212 which depends on
> php5-common 5.5.9. Ubuntu FTW.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Islandora-CLAW/CLAW/pull/201#issuecomment-215136984>
>



-- 
Benjamin Rosner

Instructional Applications Developer
Library and Academic Information Services
Sulzberger Hall Annex, Barnard College
p: 212-854-9005

**Comment by DiegoPino** [comment]
Ubuntu 16.04 will not work with php5-stomp. It's Php 7. There are some fixes in place to make stomp 7 compliand but we will need to compile from source.
https://bugs.php.net/bug.php?id=71445 -> php stomp does not work out of the box. 
But we can tune it later to use for example
https://github.com/reactphp/stomp wich works under PHP7?

**Comment by whikloj** [comment]
I think Ubuntu 16.04 will have both PHP5 and PHP7 as options. This is (I think) because the stomp package has not been added to the 5.6 PPA.
https://github.com/oerdnj/deb.sury.org/issues/332

**Comment by whikloj** [comment]
So maybe we revert to PHP 5.5 for now and worry about it when we get to Ubuntu 16.04.

I try to open a revert PR.

**Comment by oerdnj** [comment]
JFTR `php-stomp` from `ppa:ondrej/php` has both PHP 5.6 and PHP 7.0 support (and lately also PHP 5.5 support, see https://www.patreon.com/posts/php-5-5-hit-ppa-5222006).

**Comment by whikloj** [comment]
Thanks @oerdnj.

@ruebot, our problem is `drush`

```
vagrant@islandora-deux:~$ apt-cache depends drush
drush
  Depends: php5-cli
  Depends: php-console-table
  Suggests: git
  Suggests: mysql-client
    mysql-client-5.5
  Suggests: drupal7
 |Recommends: wget
  Recommends: curl
 |Recommends: php5-mysql
  Recommends: php5-pgsql
  Conflicts: <drush-make>
  Replaces: <drush-make>
```

So we need to change the way and the version we are installing as it pulls php 5.5 in with it after I had installed php 5.6.

**Comment by oerdnj** [comment]
@whikloj 

I did some updates to the `ppa:ondrej/php` that allow coinstalling with old `src:php5` package last night: https://www.patreon.com/posts/php-7-0-now-with-5221987

So in theory, drush pulling php5-cli should not be a problem.

**Comment by whikloj** [comment]
You are right @oerdnj, I switched anyways as the default drush is very old. The real problem was I forgot to install php5.6-xml. 😞

**Pull request merged**